### PR TITLE
IBC version negotiation

### PR DESCRIPTION
**Pull request by michaelfig**
Make the network protocol add negotiated metadata to the local (listening) or remote (requested) address when a connection is complete.  This makes it possible to learn, say, IBC version information from the connection before sending or receiving any packets.

Also, make the IBC connection states clearer.

### CODE REVIEW

File: packages/SwingSet/src/vats/network/network.js [hunk 1]
```diff
@@ -568,14 +637,21 @@ export function makeLoopbackProtocolHandler(E = defaultE) {
...
+          .onInbound(lport, remoteAddr, localAddr, lhandler)
+          .catch(e => rethrowUnlessMissing(e)));
+
+      if (remoteSuffix) {
+        remoteAddr = `${remoteAddr}/${remoteSuffix}`;
+      }
+
+      const rchandler =
+        /** @type {ConnectionHandler} */
+        (await E(lhandler).onAccept(lport, remoteAddr, localAddr, lhandler));
```
Comment (kriskowal) [inline packages/SwingSet/src/vats/network/network.js:652]: I’m surprised the linter didn’t strip the outer parentheses.

**Review by kriskowal — APPROVED** [review]
Looks good to me.

File: packages/SwingSet/src/vats/network/network.js (hunk 1 above)
Comment (michaelfig) [reply packages/SwingSet/src/vats/network/network.js:652]: It didn't because it's a legitimate syntax for jsdoc typecast.

Comment (michaelfig) [reply packages/SwingSet/src/vats/network/network.js:652]: It didn't because it's a legitimate syntax for jsdoc typecast.

Comment (kriskowal) [reply packages/SwingSet/src/vats/network/network.js:652]: Oh, the parentheses are affiliated with the comment and eslint knows about this? That’s intriguing.

**Pull request merged**